go.mod

```diff
@@ -1,13 +1,15 @@
 module github.com/metal-stack/gardener-extension-provider-metal
 
-go 1.22
+go 1.22.0
+
+toolchain go1.23.0
```

Comment:
either or

Reply:
Update to 1.23 does not work. :(

Reply:
Why ? And which version is then taken by the go build action ? Confusing

Reply:
The code generation breaks with Go 1.23, everything else works:

```
Generating deepcopy funcs
W0829 07:01:00.632064    3119 parse.go:863] Making unsupported type entry "any" for: &types.Alias{obj:(*types.TypeName)(0xc000098be0), orig:(*types.Alias)(0xc00003c180), tparams:(*types.TypeParamList)(nil), targs:(*types.TypeList)(nil), fromRHS:(*types.Interface)(0xc000098b40), actual:(*types.Interface)(0xc000098b40)}
W0829 07:01:00.632494    3119 parse.go:863] Making unsupported type entry "reflect.uncommonType" for: &types.Alias{obj:(*types.TypeName)(0xc00270c910), orig:(*types.Alias)(0xc002800b40), tparams:(*types.TypeParamList)(nil), targs:(*types.TypeList)(nil), fromRHS:(*types.Named)(0xc000025c00), actual:(*types.Named)(0xc000025c00)}
Generating defaulters
W0829 07:01:01.430456    3153 parse.go:863] Making unsupported type entry "any" for: &types.Alias{obj:(*types.TypeName)(0xc000148b90), orig:(*types.Alias)(0xc000128100), tparams:(*types.TypeParamList)(nil), targs:(*types.TypeList)(nil), fromRHS:(*types.Interface)(0xc000148af0), actual:(*types.Interface)(0xc000148af0)}
W0829 07:01:01.430939    3153 parse.go:863] Making unsupported type entry "reflect.uncommonType" for: &types.Alias{obj:(*types.TypeName)(0xc002096690), orig:(*types.Alias)(0xc0023b2e80), tparams:(*types.TypeParamList)(nil), targs:(*types.TypeList)(nil), fromRHS:(*types.Named)(0xc00071b730), actual:(*types.Named)(0xc00071b730)}
WARNING: generate-internal-groups.sh is deprecated.
WARNING: Please use k8s.io/code-generator/kube_codegen.sh instead.
...
```